src/lib/tls/tls12/tls_channel_impl_12.cpp

```diff
@@ -316,6 +316,11 @@ size_t Channel_Impl_12::from_peer(std::span<const uint8_t> data) {
          BOTAN_ASSERT_IMPLICATION(epoch0_restart, allow_epoch0_restart, "Allowed state");
 
          const bool initial_record = epoch0_restart || (!pending_state() && !active_state());
+         bool initial_handshake_message = false;
+         if(record.type() == Record_Type::Handshake) {
+            Handshake_Type type = static_cast<Handshake_Type>(m_record_buf[0]);
```

Comment:
I think you need to check that `m_record_buf` is not empty before doing this. At least I'm guessing that is the cause of the CiFuzz failure.

Reply:
yes, fixed

Reply:
sorry, i am first time doing this. now it's done